lib/active_record/typed_store/dsl.rb

```
@@ -6,10 +6,34 @@ module ActiveRecord::TypedStore
   class DSL
     attr_reader :fields, :coder
 
-    def initialize(attribute_name, options)
-      @coder = options.fetch(:coder) { default_coder(attribute_name) }
-      @accessors = options[:accessors]
-      @accessors = [] if options[:accessors] == false
+    def initialize(store_name, options)
+      @coder = options.fetch(:coder) { default_coder(store_name) }
+      @store_name = store_name
+      @prefix =
+        case options[:prefix]
+        when String, Symbol
+          "#{options[:prefix]}_"
+        when TrueClass
+          "#{store_name}_"
+        else
+          ""
```

I'd rather raise an `ArgumentError` if `prefix:` is an unexpected type.